miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppWebViewClient.kt

```diff
@@ -14,10 +14,24 @@ internal class MiniAppWebViewClient(
     private val customScheme: String
 ) : WebViewClient() {
 
-    override fun shouldInterceptRequest(
-        view: WebView,
+    override fun shouldInterceptRequest(view: WebView, request: WebResourceRequest): WebResourceResponse? {
+        val interceptedWebRequest = loader.shouldInterceptRequest(request.url)
+
+        interceptMimeType(interceptedWebRequest, request)
+
+        return interceptedWebRequest
+    }
+
+    @VisibleForTesting
+    internal fun interceptMimeType(
+        interceptedWebRequest: WebResourceResponse?,
         request: WebResourceRequest
-    ): WebResourceResponse? = loader.shouldInterceptRequest(request.url)
+    ) {
+        interceptedWebRequest?.let {
+            if (request.url.toString().endsWith("js", true))
```

Comment:
Do we need to check the `null` as well? e.g. `if (request.url != null && request.url.toString()..)`

Reply:
I would put a dot `(.)` before the `"js"` for safe call.

Reply:
when the method `shouldInterceptRequest` was triggered automatically by webview, `request.url` is never null, that's what I know 😅 but not so sure.

Reply:
Yeah, I was concerned if any client send a non-null file e.g. "somethingjs", which is an invalid JS file. But, "something.js" is a valid JS file.

Reply:
Yeah, I was concerned if any client send a file e.g. "somethingjs", which is an invalid JS file. But, "something.js" is a valid JS file.

Reply:
absolutely true.